### PR TITLE
feat: new ipfs pinning proxy deployment

### PR DESCRIPTION
**Pull request by luzzif**

### CODE REVIEW

File: terraform/k8s/main.tf [hunk 1]
```diff
@@ -246,42 +262,22 @@ resource "kubernetes_deployment" "ipfs_pinner" {
...
-          image             = "luzzif/carrot-kpi-ipfs-pinner:v0.5.3"
-          image_pull_policy = "IfNotPresent"
-          env {
-            name  = "IPFS_API_ENDPOINT"
-            value = "http://ipfs-node:9095"
-          }
-          env {
-            name  = "WS_RPC_ENDPOINT"
-            value = var.ws_rpc_url_scroll_testnet
+        volume {
```
Comment (guerrap) [inline terraform/k8s/main.tf:298]: What's this volume for? Wasn't it there before?

File: terraform/k8s/main.tf [hunk 2]
```diff
@@ -446,8 +648,10 @@ resource "kubernetes_ingress_v1" "ipfs_gateway" {
     name      = "ipfs-gateway"
     namespace = kubernetes_namespace.api.metadata.0.name
     annotations = {
-      "kubernetes.io/ingress.class" = "nginx"
-      "cert-manager.io/issuer"      = "letsencrypt-prod"
+      "kubernetes.io/ingress.class"                       = "nginx"
+      "cert-manager.io/issuer"                            = "letsencrypt-prod"
+      "nginx.ingress.kubernetes.io/limit-rps"             = 10
```
Comment (guerrap) [inline terraform/k8s/main.tf:653]: Why limit to `10 rps` on the IPFS too? The service is not accessible from outside of the k8s, right?

File: packages/pinning-proxy/src/routes/token.ts [hunk 3]
```diff
@@ -111,6 +111,16 @@ export const getTokenRoute = ({
...
             }
 
+            try {
+                await deleteNonce({ client: dbClient, address });
+            } catch (error) {
+                console.error(
+                    "Error while deleting nonce from database",
+                    error
+                );
+                return internal("Error while generating JWT");
```
[the remark this answers is not in the document]
Comment (luzzif) [reply packages/pinning-proxy/src/routes/token.ts:122]: The `internal` call generates a `500` error response that Hapi sends back to the client with a given message. I don't think the error message in this case should convey those details about the internal working of the API, so I just put a generic error message here.

[the remark this answers is not in the document]
Comment (guerrap) [reply packages/pinning-proxy/src/routes/token.ts:122]: Oh ye, sounds good then :+1: .

**Review by guerrap — APPROVED** [review]

**Pull request merged**